sys/net/network_layer/fib/fib.c

```diff
@@ -609,6 +691,712 @@ int fib_get_num_used_entries(void)
     return used_entries;
 }
 
+/* source route handling */
+int fib_sr_create(fib_sr_t **fib_sr, kernel_pid_t sr_iface_id, uint32_t sr_flags,
+                  uint32_t sr_lifetime)
+{
+    mutex_lock(&mtx_access);
+    if ((fib_sr == NULL) || (sr_lifetime == 0)) {
+        mutex_unlock(&mtx_access);
+        return -EFAULT;
+    }
+
+    for (size_t i = 0; i < FIB_MAX_FIB_SR; ++i) {
+        if (fib_sr_table[i].sr_lifetime.seconds == 0
+            && fib_sr_table[i].sr_lifetime.microseconds == 0) {
+            fib_sr_table[i].sr_iface_id = sr_iface_id;
+            fib_sr_table[i].sr_flags = sr_flags;
+            fib_sr_table[i].sr_path = NULL;
+            fib_sr_table[i].sr_dest = NULL;
+            if (sr_lifetime < FIB_LIFETIME_NO_EXPIRE) {
+                fib_ms_to_timex(sr_lifetime, &fib_sr_table[i].sr_lifetime);
+            }
+            else {
+                fib_sr_table[i].sr_lifetime.seconds = FIB_LIFETIME_NO_EXPIRE;
+                fib_sr_table[i].sr_lifetime.microseconds = FIB_LIFETIME_NO_EXPIRE;
+            }
+            *fib_sr = &fib_sr_table[i];
+            mutex_unlock(&mtx_access);
+            return 0;
+        }
+    }
+
+    mutex_unlock(&mtx_access);
+    return -ENOBUFS;
+}
+
+
+static int fib_sr_check_lifetime(fib_sr_t *fib_sr, timex_t *now)
```

Comment:
I would tend to remove the argument `now` and call `vtimer_now()` within this function